src/Logic.hs

```diff
@@ -397,15 +396,22 @@ proceed state = case Pr.getIntegrationCandidate state of
 -- TODO: Get rid of the tuple; just pass the ID and do the lookup with fromJust.
 proceedCandidate :: (PullRequestId, PullRequest) -> ProjectState -> Action ProjectState
 proceedCandidate (pullRequestId, pullRequest) state =
-  case Pr.buildStatus pullRequest of
-    BuildNotStarted -> error "integration candidate build should at least be pending"
-    BuildPending    -> pure state
-    BuildSucceeded  -> pushCandidate (pullRequestId, pullRequest) state
-    BuildFailed     -> do
-      -- If the build failed, this is no longer a candidate.
-      leaveComment pullRequestId "The build failed."
+  case Pr.integrationStatus pullRequest of
```

Comment:
The function `proceed` has this comment at the call site for `proceedCandidate`

> ```haskell
>  -- If there is a candidate, nothing needs to be done. TODO: not even if the
>  -- build has finished for the candidate? Or if it has not even been started?
>  -- Do I handle that here or in the build status changed event? I think the
>  -- answer is "do as much as possible here" because the events are ephemeral,
>  -- but the state can be persisted to disk, so the process can continue after a
>  -- restart.
> ```

What this function does doesn't look like "nothing", so I suppose that comment needs an update.

Reply:
Good find. I removed the entire comment, the approach of “do as much as possible” in the to do comment is already the approach taken by `proceedCandidate`.